Make `id` optional in all components

mbohal opened the pull request:
This is now possible after https://github.com/visionappscz/eslint-config-visionapps/pull/7 was merged.

adamkudrna reviewed (approved):
- I'd change PR title from `Make HTML attribute id optional in all elements` to `Make `id` optional in all components`.
- I'm not sure about dumping the `htmlFor` attribute—why not use the `id` when it's there… But it's a subtle detail I'm ready to leave as is.

mbohal commented:
@adamkudrna 
> 'm not sure about dumping the htmlFor attribute—why not use the id when it's there

Good point, I have fixed that.

Also I have removed the now unnecessary `id` attributes from the `README.mdx` files. I think the less cluttered the examples are, the better.
